### ISSUE TITLE
[DOC] - Update docs about certificates

### ISSUE THREAD

**Issue by viniciusdc** [nebari-dev/nebari-docs#421]
### Preliminary Checks

- [X] This issue is not a question, feature request, RFC, or anything other than a bug report. Please post those things in GitHub Discussions: https://github.com/nebari-dev/nebari/discussions

### Summary

- Update information regarding local deployment and the use of /etc/hosts when using lets-encrypt
- Due to https://github.com/nebari-dev/nebari/pull/2352, we need to tell users that if the domain is updated, the cert.json file present in the PV needs to be deleted so that traefik generates new certs for the new domain

### Steps to Resolve this Issue

1. Update docs
2.
3.
...

**Comment by viniciusdc** [comment]
Just a bit more of context to this:

The current documentation needs updates to address changes in how certificates are managed during local deployments, considering the changes made by now having persistency within the generated self-signed certs, especially in the context of using Let's Encrypt with Traefik Ingress and how `/etc/hosts` interacts with all this. Recent experiences suggest revising guidelines to avoid confusion and helpful information when encountering issues.

## Details & Context

1. `/etc/hosts` and Certificate Generation
Previously, mapping a DNS name in the `/etc/hosts` to MetalLB's external IP was sufficient for Let's Encrypt (using the production ACME server) to generate a valid certificate. This was mainly because, in the Qhub's era, when developers attempted to deploy it locally, we pinned the range for the external LB IP to always point to `172.18.1.100`, which, by that time, we used to have a Cloudflare DNS pointing to it. Its first appearance could be tracked down roughly to [here](https://github.com/nebari-dev/nebari/blob/53c95d17582f3fb4e7f8750bc2a901f795f44cae/docs/source/06_developers_contrib_guide/04_tests.md?plain=1#L238-L250). And since then this mostly was untouched.

2. Persistent Volume Claims and Certificate Renewal:
- There have been instances where, despite redeploying a cluster, Traefik continued to utilize an outdated `cert.json` file from a persistent volume (PV), instead of generating a new certificate. This could get a note section in the Cers docs as a possible workaround for the user in case such a situation occurs -- this is not a common thing because we don't expect users to change the whole domain name often. 


## In conclusion:

- Remove outdated references to /etc/hosts setup when in the context of Let's Encrypt certifications to avoid confusion.
- If users are looking for why their domain is interested by their browsers when using LetsEncrypt, a mention to what their SSL certs are not recognized should be made, and pointing to the DNS docs.
- For users who only want to test/dev the deployment and would like to avoid this DNS route, we should direct them to using custom certs instead and opting for [openssl](https://letsencrypt.org/docs/certificates-for-localhost/#making-and-trusting-your-own-certificates) or [certgen](https://github.com/minio/certgen) to help in this endeavor

**Comment by aktech** [comment]
People can general certs for local deployment too with DNS challenge, see this comment: https://github.com/nebari-dev/nebari/issues/1405#issuecomment-2139417252

I would be nice to add docs for that to patch it for local deployment and eventually implement an option to generate certs via dns challenge.

**Comment by aktech** [comment]
Note regarding certs for local deployment

I saw an issue with terminal on the local deployment, it doesn't loads. (It might just be on my local deployment) I see the terminal websocket connection to terminal endpoint fails.

https://github.com/nebari-dev/nebari-docs/assets/5647941/52049e0e-78d6-4270-9a02-2cb486a7eafc
![Screenshot 2024-05-29 at 4 59 47 pm](https://github.com/nebari-dev/nebari-docs/assets/5647941/692b083c-003d-44ea-818c-7c0879ad58dc)

It worked when nebari is running with valid certs (I generated it via DNS challenge as mentioned above)

**Comment by kcpevey** [comment]
@aktech / @viniciusdc does https://github.com/nebari-dev/nebari-docs/pull/470 sufficiently cover this such that this can be closed now?

**Comment by viniciusdc** [comment]
Hi @kcpevey, the above mentioned PR was enough for this issue. I will be closing it now, the above comments from @aktech was due to some experimental changes we are implementing on nebari that will be interesting to document later on (I just asked him to add some notes here so that we didn't forget) -- I will create a new issue for that :)
